opentracing-noop/src/main/java/io/opentracing/NoopScheduler.java

```diff
@@ -0,0 +1,37 @@
+package io.opentracing;
+
+/**
+ * A noop (i.e., cheap-as-possible) implementation of a Scheduler.
+ */
+public class NoopScheduler implements Scheduler {
+    public static final Continuation NOOP_CONTINUATION = new Continuation();
+
+    @Override
+    public Span active() {
+        return null;
+    }
+
+    @Override
+    public Continuation capture(Span span) {
+        return NOOP_CONTINUATION;
+    }
+
+    @Override
+    public SpanContext activeContext() {
+        return null;
+    }
+
+    @Override
+    public Continuation captureActive() { return NOOP_CONTINUATION; }
+
+    public static class Continuation implements Scheduler.Continuation {
+        @Override
+        public Span activate(boolean finishOnDeactivate) { return null; }
```

Comment:
> There are alternatives, though they would involve changes to the contract for captureActive – namely that a new Span must be created if none was active, though then we need an operation name and so forth.

Why not replace `null` with `NoSpan` (not the same as noop)? In most cases it will simplify the usage there is no need to plague the app/library code with null-checks.

It has the downside that you never know whether you actually have a real span or not, and that might require adding one extra function to the Span interface to know for sure, but imho that's a fair price to pay for a simpler and safer usage experience.